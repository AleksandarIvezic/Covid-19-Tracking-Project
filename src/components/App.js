import React, { useEffect } from 'react';
import '../styling/App.css';
import { Route, Switch } from 'react-router';
import { useSelector, useDispatch } from 'react-redux';
import Navbar from './Navbar';
import Countries from './Countries';
import CountryDetails from './CountryDetails';
import countryIso from '../libraries/countryIso';
import { loadCountriesThunk, selectCountry, isLoading } from '../Redux/countries/countries';

function App() {
  const countries = useSelector((state) => state.countriesReducer.countries);
  const total = useSelector((state) => state.countriesReducer.total);
  const loading = useSelector((state) => state.countriesReducer.loading);
  const currentCountry = useSelector((state) => state.countriesReducer.currentCountry);
  const iso = countryIso[currentCountry];
  const imgLink = iso ? `https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/${iso.toLowerCase()}/vector.svg` : '/World_map.png';
  const dispatch = useDispatch();

  const handleClick = (country) => dispatch(selectCountry(country));

  useEffect(() => {
    dispatch(isLoading());
    dispatch(loadCountriesThunk());
  }, [dispatch]);
  return (
    <div className="App">
      <Navbar />
      <Switch>
        <Route exact path="/">
          <Countries
            countries={countries}
            total={total}
            loading={loading}
            handleClick={handleClick}
          />
        </Route>
        <Route path="/country">
          <CountryDetails current={currentCountry} img={imgLink} />
        </Route>
      </Switch>
    </div>
  );
}

export default App;
