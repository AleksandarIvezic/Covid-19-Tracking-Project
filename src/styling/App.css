.App {
  text-align: center;
  background-color: #4369b2;
  color: #fff;
}
