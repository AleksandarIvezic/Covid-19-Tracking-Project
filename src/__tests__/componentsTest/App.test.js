import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/storeConfig';
import App from '../../components/App';

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
