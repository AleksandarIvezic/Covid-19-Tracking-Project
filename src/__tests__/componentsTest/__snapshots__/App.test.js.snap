// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="d-flex justify-content-between p-1 sticky-top"
  >
    <span
      className="fw-bold text-uppercase back"
    >
      Countries
    </span>
    <span
      className="fw-light fs-6 text-lowercase"
    >
      Tracking
      /
    </span>
    <div>
      <i
        className="fas fa-microphone mx-2"
      />
      <i
        className="fas fa-cog mx-2"
      />
    </div>
  </nav>
  <div>
    <div
      className="header d-flex p-1"
    >
      <img
        alt="world-map"
        className="c-img"
        height="50%"
        src="World_map_svg.svg"
        width="50%"
      />
      <div
        className="right-side d-flex flex-column align-items-center justify-content-center w-50 fw-bold"
      >
        <h1
          className="m-0 fs-3 lh-1 fw-bolder"
        >
          WORLD
        </h1>
        <span
          className="lh-1"
        >
          NaN
        </span>
        <span
          className="lh-1"
        >
          total cases
        </span>
      </div>
    </div>
    <form
      className="d-flex stripe-text justify-content-between"
    >
      <span
        className="m-1"
      >
        STATS BY COUNTRY
      </span>
      <input
        className="search"
        name="search"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
    </form>
    <img
      alt="loader"
      className="loading-img"
      src="virus-img.png"
    />
    <ul
      className="d-flex flex-wrap"
    />
  </div>
</div>
`;
